Sources/MongoSwift/ClientSession.swift

```diff
@@ -331,12 +331,18 @@ public final class ClientSession {
             throw ClientSession.SessionInactiveError
         }
 
+        guard let bson = bson_new() else {
+            fatalError("failed to allocate bson_t")
+        }
+        defer { bson_destroy(bson) }
+
         var error = bson_error_t()
-        try doc.withMutableBSONPointer { docPtr in
```

Comment:
I opted to only add the immutable version, since I don't think there would be any way for us to track the writer index properly if we exposed a mutable pointer. This is the only place that is using `withMutableBSONPointer` that would need to continue to do so once we swapped out libraries, so I just added a manual implementation that drops into libbson.